### ISSUE TITLE
AddEmbeddedPackagePart is saved to wrong path

### ISSUE THREAD

**Issue by Asbjoedt** [dotnet/Open-XML-SDK#1318]
**Describe the bug**

_ISSUE 1_

You can iterate all EmbeddedPackageParts by doing the below. It iterates WorksheetParts and then each EmbeddedPackagePart.
```
using (SpreadsheetDocument spreadsheet = SpreadsheetDocument.Open(filepath, true))
{
    IEnumerable<WorksheetPart> worksheetParts = spreadsheet.WorkbookPart.WorksheetParts;
    foreach (WorksheetPart worksheetPart in worksheetParts)
    {
        IEnumerable<EmbeddedPackagePart> embedpackages = worksheetPart.EmbeddedPackageParts;
        foreach (EmbeddedPackagePart part in embedpackages)
        {

        }
    }
}
```
You can delete any EmbeddedPackagePart by deleting it at the WorksheetPart level like this.
```
worksheetPart.DeletePart(part);
```

However, if you want to AddEmbeddedPackagePart you can do so also at the WorksheetPart level:
```
EmbeddedPackagePart embedpack = worksheetPart.AddEmbeddedPackagePart(contentType);
```
But the new EmbeddedPackagePart will be saved to this URI where a new ```embeddings``` folder will be created:
```
xl/worksheets/embeddings/package.bin
```
If you use Excel, all EmbeddedPackageParts are saved in this folder instead:
```
xl/embeddings/package.bin
```

Is this correct behaviour? Should Open XML SDK also save the part to ```xl/embeddings```?

_ISSUE 2:_

Excel saves the EmbeddedPackagePart with original extension i.e. ```.xlsx```.

Open XML SDK saves EmbeddedPackagePart with ```.bin``` extension.

**Observed behavior**
1. EmbeddedPackagePart is saved to ```xl/worksheets/embeddings```

2. EmbeddedPackagePart is saved with ```.bin``` extension.

**Expected behavior**
1. EmbeddedPackagePart should be saved to ```xl/embeddings```

2. EmbeddedPackagePart should be saved with the original extension of the package i.e. ```.xlsx```.

**Desktop (please complete the following information):**
* OS: Windows 11
* Office version: 2212
* .NET Target: .Net latest version
* DocumentFormat.OpenXml Version: 2.19

**Comment by Asbjoedt** [comment]
Hi @twsouthwick 

I made a mistake trying to PR to v2.20 so I pulled it again. I think you may want my PR for v3 or another version than 2.20.

You can look at my code in the closed PR.
Let me know, if you want it and I can PR in the branch you want it.

**Comment by twsouthwick** [comment]
Hi @Asbjoedt, I would prefer any feature requests against main and then we can decide if it's worth it to back port.

This issue looks to be a schema issue similar to #1292.

/cc @tomjebo @hunyu @mikeebowen

**Comment by tomjebo** [comment]
@Asbjoedt do you have test code including the embedded contents to be added? I will need to verify all the behavior here. 

1. I would have to test this to verify the behavior but I suspect that Excel would be ok with the path to the part as long as the relationship pointed correctly to where it lives. The ISO 29500 standard and our implementer notes don't specify a path. They give examples but they specify that the package must be the target of the ...relationships/package relationship. If it does need correction because Excel doesn't accept what the SDK uses or we just decide it really has to match Excel's address, we do this on the backend/generation phase. If Excel (or Word or PowerPoint) don't care what the path is, then we could use a feature on part creation to allow the coder to specify a new path.

2. As far as the .xlsx, we shouldn't use .bin as this is an Office Open XML package and not an OLE object. This would also be done on the backend/generation phase in our package.xml CT_EmbeddedPackagePart definition.

**Comment by Asbjoedt** [comment]
@tomjebo

1. I think imitating Excel to the widest possible extent as long as it does not go against the OOXML standards, is best. It is confusing to have to look for embeddings multiple places based on whether you used Excel or Open XML SDK. However, you are right that technically as long as the relationship paths are correctly written, Excel can read the data.

2. I agree. As a solution, I suggest to imitate the ImagePartType, that is already in the codebase and I reused this code yesterday for a fix for embedded packages, do take a look and compare it with the ImagePartType stuff: https://github.com/dotnet/Open-XML-SDK/commit/9f7ed9700b5a45f45a956d5aeee4e1c77d13bbce
I am not fully sure, it completes the fix, but it goes a long way to fix it.

Here's code for you to test the behaviour on .xlsx files.
```
public void EmbedObject(string filepath)
{
    List<EmbeddedPackagePart> packages = new List<EmbeddedPackagePart>();

    using (SpreadsheetDocument spreadsheet = SpreadsheetDocument.Open(filepath, true))
    {
        IEnumerable<WorksheetPart> worksheetParts = spreadsheet.WorkbookPart.WorksheetParts;
        foreach (WorksheetPart worksheetPart in worksheetParts)
        {
            packages = worksheetPart.EmbeddedPackageParts.Distinct().ToList();
            foreach (EmbeddedPackagePart part in packages)
            {
                // Create new EmbeddedPackage of contentType .xlsx
                EmbeddedPackagePart new_EmbeddedPackagePart = worksheetPart.AddEmbeddedPackagePart("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet");

                // Feed data to the new package from an .xlsx file
                var new_Stream = new MemoryStream();
                using (FileStream file = new FileStream(filepath_to_embed, FileMode.Open, FileAccess.Read))
                {
                    file.CopyTo(new_Stream);
                }
                new_EmbeddedPackagePart.FeedData(new_Stream);
                new_Stream.Dispose();
            }
        }
    }
}
```
I recommend you create your own test sample, where you i.e. create two .xlsx spreadsheets and embed one of the .xlsx into the other as an object. You should try the embedding process first using Excel and then Open XML SDK.

**Comment by tomjebo** [comment]
@Asbjoedt thanks for the additional details and explanation. I can create a sample to test this and verify the behavior. Regarding (1) above, yes, I agree that we should imitate Excel also. The SDK is not only built around the ISO standard but around the Office behaviors. So good point.

**Comment by Asbjoedt** [comment]
@tomjebo. It would be great if this fix could get into 2.20, if possible.

**Comment by tomjebo** [comment]
@Asbjoedt I'm working on getting this into v2.20.0.

**Comment by twsouthwick** [comment]
@tomjebo this was included in v2.20.0 right? Can this be closed?

**Comment by tomjebo** [comment]
Before closing this I wanted to circle back to explain something about the path.  I was looking at this and I see why `EmbeddedPackagePart`s are saved to a new `embeddings` folder in the folder of the referencing part. `EmbeddedPackagePart` is a shared type in ISO 29500-1. That means that it is a target of an explicit relationship of several parts in WordprocessingML, SpreadsheetML and PresentationML documents. However, these documents don't have the same structures wrt where those referencing parts live path-wise.  So when the SDK designers were considering how to declaratively specify using XML definitions with a targetPath specification like `embeddings` or "../embeddings" or "/embeddings", they couldn't come up with one that would satisfy all three apps. For Excel, it would be "../embeddings" but for Word it would be "./
